File: 217-contains-duplicate/217-contains-duplicate.java
class Solution {
    public boolean containsDuplicate(int[] nums) {
        int n = nums.length;
        Map<Integer, Integer> store = new HashMap<>();
        
        for(int i = 0; i < nums.length; i++)
        {
            Integer c = store.get(nums[i]);
            
            if(c != null && c == 1) {
                return true;
            }
            else {
                c = 0;
                store.put(nums[i], c + 1);
            }
        }
        
        return false;
    }
}

File: 217-contains-duplicate/217-contains-duplicate.cpp
class Solution {
public:
    bool containsDuplicate(vector<int>& nums) {
        
        map<int, int> duplicate; 
        
        for(int num: nums) {
            if(duplicate.find(num) != duplicate.end()) {
                return true;
            }
            duplicate[num] = 1;
        }
        
        return false;
    }
};